kustomize/components/clairpostgres/postgres.statefulset.yaml

```diff
@@ -1,15 +1,14 @@
 apiVersion: apps/v1
-kind: Deployment
+kind: StatefulSet
 metadata:
   name: clair-postgres
   labels:
     quay-component: clair-postgres
   annotations:
     quay-component: clair-postgres
 spec:
+  serviceName: clair-postgres
   replicas: 1
-  strategy:
-    type: Recreate
```

Comment:
This will switch the deployment to a rolling upgrade, which could cause the old and new pod to run at the same time. Will that cause issues?

Reply:
Since it is a stateful set, it will only run one pod at a time

Reply:
In theory, issues can occur where Quay is operating on a Clair that has a split-brain during a deployment (i.e. the security worker requests the indexing state from a new pod, sees it has changed then makes an indexing request but it goes to an old pod). In practice, this is how it works in production and we've never seen problems so I think it's probably fine but something to document.